connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestServer.java

```diff
@@ -309,7 +309,11 @@ void registerRestExtensions(HerderProvider provider, Plugins plugins, ResourceCo
             config, ConnectRestExtension.class);
 
         long herderRequestTimeoutMs = ConnectorsResource.REQUEST_TIMEOUT_MS;
-        Integer rebalanceTimeoutMs = config.getInt(DistributedConfig.REBALANCE_TIMEOUT_MS_CONFIG);
+
+        Integer rebalanceTimeoutMs =
+            config.typeOf(DistributedConfig.REBALANCE_TIMEOUT_MS_CONFIG) == null ? null :
+            config.getInt(DistributedConfig.REBALANCE_TIMEOUT_MS_CONFIG);
+
```

Comment:
Would it be better to get the value only when the return type is an integer rather than do that when it the type is non-null? For example, something like:
```
Integer rebalanceTimeoutMs = null;
if (config.typeOf(DistributedConfig.REBALANCE_TIMEOUT_MS_CONFIG) == ConfigDef.Type.INTEGER) {
    rebalanceTimeoutMs = config.getInt(DistributedConfig.REBALANCE_TIMEOUT_MS_CONFIG);
}
```
This may be a bit easier to read, and it's also going to cast to an integer only if the value actually is an integer.

Reply:
@rhauch, fair point. I was really hoping for `AbstractConfig::hasKey`` method, but the `typeOf` was the closest I could get.

There are two reasons I prefer my method: 
 * The ternary is a one-liner so it's concise and compact assignation. If it were already the pattern in the package, I would have even take a step further and add `final`.
 * If the config is changed to a non-integer type, I would prefer the unit test to fail rather than carry on and assign `null`. For example, if the timeout config is changed to type `Long`, all of our tests would continue to pass, and we'd silently start changing behavior unexpectedly.

Reply:
To your second bullet point, the existing code won't work if the config type is changed to a long (for example), because `getInt(...)` is basically just `return (Integer) get(key);`, which will result in a class cast exception if the value is anything but an integer. Besides, any change of the config type would require changing all of the places where `getInt(DistributedConfig.REBALANCE_TIMEOUT_MS_CONFIG)` is called.

To your first point, I agree that the current code is pretty readable, but becomes less readable if you were to change the existing code to only convert if the config value were an integer, such as:
```
Integer rebalanceTimeoutMs =
    config.typeOf(DistributedConfig.REBALANCE_TIMEOUT_MS_CONFIG) == ConfigDef.Type.INTEGER ? 
    config.getInt(DistributedConfig.REBALANCE_TIMEOUT_MS_CONFIG) :
    null;
```
or
```
Integer rebalanceTimeoutMs =
    config.typeOf(DistributedConfig.REBALANCE_TIMEOUT_MS_CONFIG) != ConfigDef.Type.INTEGER ? null :
    config.getInt(DistributedConfig.REBALANCE_TIMEOUT_MS_CONFIG);
```
The latter makes it really hard to miss the `!=`.

Reply:
@rhauch I'm not sure we're on the same page: I think it is an advantage of the PR as it is to fail on the Integer typecasting. This way, unit tests will fail and we will be alerted. In the case where we test against `INTEGER` type, we will see no failures, which I'd argue is a disadvantage: Behavior would silently change.

To your point about "any change of the config type would require changing all of the places", that is true, and I think the code as its written in the PR helps make sure we catch this case, by virtue of a failing unit test.

Maybe I'm still not understanding your reasoning?

Reply:
How about we get rid of the problem altogether -- define `WorkerConfig.rebalanceTimeout()` that returns null by default, and then override it in `DistributedConfig` to return `getInt(DistributedConfig.REBALANCE_TIMEOUT_MS_CONFIG)` since that will always be an integer. The latter's method is trivial, and the code in `RestServer` becomes simpler and easier to read.

I think there's enough precedence in `StreamsConfig` and `ConverterConfig` to add getter methods (without `get` prefix). The fact that it cleans this up this significantly is also good justification.

Reply:
@rhauch sounds good!